client/src/components/pages/application/Signup/index.js

```diff
@@ -1,5 +1,195 @@
 import React from 'react';
+import axios from 'axios';
+import Alert from '../../../common/Alert';
+import Card from '../../../common/card';
+import Typography from '../../../common/Typography';
+import TextField from '../../../common/TextField';
+import Button from '../../../common/Button';
+import signupValidate from '../../../../utils/application/SignupValidation';
+import SelectBox from '../../../application/SelectBox';
+import signupImage from '../../../../assets/images/signup.svg';
+import './index.css';
 
-const SignUp = () => <h1>SignUp</h1>;
+const initialState = {
+  fullName: '',
+  email: '',
+  password: '',
+  passwordConfirmation: '',
+  location: '',
+  message: [],
+};
+
+class SignUp extends React.Component {
+  state = initialState;
+
+  handleSignIn = () => {
+    const { history } = this.props;
+    history.push('/login');
+  };
```

Comment:
here you have to add history to props at the bottom of the class because there is eslint problem says that : 
'history' is missing in props validation so you should add props like this :
```js
Signup.propTypes = {
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};
```
and you can only call push without history unless you extract it like this 
```js
 const { history: { push } } = this.props;
```
so then just call it by this way easier and you can call it every where in your class
```js
push('/login');
```

Reply:
thanks a lot, this one is very usefull for me.